wasmapi/rust/src/rust_bindings/map.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::rust_bindings::helpers::{any_to_buf_ptr, string_to_buf_ptr};
+
+extern "C" {
+    #[link_name = "newMap"]
+    fn new_map(name: u64, typ: u32, key_size: u32, value_size: u32, max_entries: u32) -> u32;
+    #[link_name = "getMap"]
+    fn get_map(name: u64) -> u32;
+    #[link_name = "mapLookup"]
+    fn map_lookup(map: u32, key_ptr: u64, value_ptr: u64) -> u32;
+    #[link_name = "mapUpdate"]
+    fn map_update(map: u32, key_ptr: u64, value_ptr: u64, flags: u64) -> u32;
+    #[link_name = "mapDelete"]
+    fn map_delete(map: u32, key_ptr: u64) -> u32;
+    #[link_name = "mapRelease"]
+    fn map_release(map: u32) -> u32;
+}
+
+#[repr(u32)]
+#[derive(Debug, Clone, Copy)]
+pub enum MapType {
+    Unspecified = 0,
+    Hash,
+    Array,
+    ProgramArray,
+    PerfEventArray,
+    PerCPUHash,
+    PerCPUArray,
+    StackTrace,
+    CGroupArray,
+    LRUHash,
+    LRUCPUHash,
+    LPMTrie,
+    ArrayOfMaps,
+    HashOfMaps,
+    DevMap,
+    SockMap,
+    CPUMap,
+    XSKMap,
+    SockHash,
+    CGroupStorage,
+    ReusePortSockArray,
+    PerCPUCGroupStorage,
+    Queue,
+    Stack,
+    SkStorage,
+    DevMapHash,
+    StructOpsMap,
+    RingBuf,
+    InodeStorage,
+    TaskStorage,
+    BloomFilter,
+    UserRingbuf,
+    CgroupStorage,
+    Arena,
+}
+
+pub type Result<T> = std::result::Result<T, MapError>;
+
+pub enum MapError {
+    ErrCreateMap(String),
+    ErrNotFound(String),
+    ErrLookup(String),
+    ErrUpdateMap(String),
+}
+
+#[repr(u64)]
+#[derive(Clone, Copy)]
+pub enum MapUpdateFlags {
+    UpdateAny = 0,
+    UpdateNoExist = 1,
+    UpdateExist = 2,
+    UpdateLock = 4,
+}
+
+#[derive(Copy, Clone)]
```

Comment:
Do we really want to allow copying and cloning maps? Perhaps this shouldn't be allowed since those refer to a object handled by the host. 

I'd need to run some experiments to see how it behaves though.